### PR TITLE
BOAC-2421: Fills gaps in expected grad terms

### PR DESCRIPTION
**Pull request by lyttam**
https://jira.ets.berkeley.edu/jira/browse/BOAC-2421

### CODE REVIEW

File: boac/merged/cohort_filter_options.py [hunk 1]
```diff
@@ -394,8 +394,10 @@ def _genders():
 
 
 def _grad_terms():
-    term_ids = [r['expected_grad_term'] for r in data_loch.get_expected_graduation_terms()]
-    return [{'name': term_name_for_sis_id(term_id), 'value': term_id} for term_id in term_ids]
+    earliest_term_id = data_loch.get_min_expected_graduation_term()['term']
+    latest_term_id = add_terms(future_term_id(), 12)
+    term_ids = term_ids_range(earliest_term_id, latest_term_id)
+    return [{'name': ' '.join(term_name_for_sis_id(term_id).split()[::-1]), 'value': term_id} for term_id in term_ids]
```
Comment (lyttam) [inline boac/merged/cohort_filter_options.py]: Reversing term name for readability, so "Summer 2019" becomes "2019 Summer".

File: boac/merged/cohort_filter_options.py [hunk 2]
```diff
@@ -394,8 +394,10 @@ def _genders():
 
 
 def _grad_terms():
-    term_ids = [r['expected_grad_term'] for r in data_loch.get_expected_graduation_terms()]
-    return [{'name': term_name_for_sis_id(term_id), 'value': term_id} for term_id in term_ids]
+    earliest_term_id = data_loch.get_min_expected_graduation_term()['term']
+    latest_term_id = add_terms(future_term_id(), 12)
```
Comment (paulkerschen) [inline boac/merged/cohort_filter_options.py]: @lyttam Are we confident that now-plus-12-terms will always cover the required range? I get the logic, but sometimes what's in the database defies logic.

Comment (lyttam) [reply boac/merged/cohort_filter_options.py]: @paulkerschen I'm not confident about it, and it's definitely up for debate. I could pick a higher number or make it configurable.

Comment (paulkerschen) [reply boac/merged/cohort_filter_options.py]: @lyttam What if you just selected the max from the database, same as you're selecting the min for the start of the interval?

Comment (lyttam) [reply boac/merged/cohort_filter_options.py]: @paulkerschen everyone agrees that's a good idea! I'll amend.

**Comment by lyttam** [comment]
@paulkerschen All amended now.

**Comment by paulkerschen** [comment]
Thanks, @lyttam. Term math!

**Pull request merged**